packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1912,6 +1918,33 @@ def last_timestamp(self) -> datetime.datetime:
     def setup(self) -> None:
         """Set up the behaviour."""
         self._schedule_round(self.initial_round_cls)
+        self._termination_round = TerminationRound(
+            self._initial_synchronized_data,
+            self.consensus_params,
+            None,
+        )
+
+    def _get_transaction_settlement_initial_round(
```

Comment:
This is a problem. The abstract_round_abci cannot know anything about this skill. What we need to do instead is to extend the FSM spec to include the mapping from the TerminalRound to the user defined FSM (to begin with `transaction_settlement_abci`).